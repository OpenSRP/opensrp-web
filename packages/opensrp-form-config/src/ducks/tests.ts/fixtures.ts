export const fixManifestReleases = [
    {
        identifier: '1.0.11',
        json: '{"forms_version":"1.0.11","identifiers":["blood_screening.json"}',
        appId: 'org.smartregister.reveal',
        appVersion: '1.2.14',
        createdAt: '2020-06-08T15:00:38.413+02:00',
        updatedAt: '2020-06-08T15:00:38.413+02:00',
    },
    {
        identifier: '1.0.12',
        json: '{"forms_version":"1.0.12","identifiers":["blood_screening.json"}',
        appId: 'org.smartregister.reveal',
        appVersion: '1.2.14',
        createdAt: '2020-06-08T15:03:00.950+02:00',
        updatedAt: '2020-06-08T15:03:00.950+02:00',
    },
    {
        identifier: '1.0.16',
        json: '{"forms_version":"1.0.16","identifiers":["blood_screening.json"}',
        appId: 'org.smartregister.reveal',
        appVersion: '1.2.16',
        createdAt: '2020-06-09T14:40:04.189+02:00',
        updatedAt: '2020-06-09T14:40:04.189+02:00',
    },
];
