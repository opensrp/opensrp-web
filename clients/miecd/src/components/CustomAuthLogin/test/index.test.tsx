import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import CustomOauthLogin from '..';
import { providers } from './fixtures';

describe('CustomOAuth login', () => {
    it('must render without crashing', () => {
        const props = {
            providers,
        };
        shallow(<CustomOauthLogin {...props} />);
    });
    it('must render correctly', () => {
        const props = {
            providers,
        };
        const wrapper = mount(<CustomOauthLogin {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
