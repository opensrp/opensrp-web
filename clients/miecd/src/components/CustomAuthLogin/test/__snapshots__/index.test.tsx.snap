// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomOAuth login must render correctly 1`] = `
<CustomOauthLogin
  ProviderLinksComponent={[Function]}
  providers={
    Object {
      "onadata": Object {
        "accessTokenUri": "https://stage-api.ona.io/o/token/",
        "authorizationUri": "https://stage-api.ona.io/o/authorize/",
        "clientId": "hunter2",
        "redirectUri": "https://example.com/oauth/callback/onadata/",
        "scopes": Array [
          "read",
          "write",
        ],
        "state": "abc",
        "userUri": "https://stage-api.ona.io/api/v1/user.json",
      },
    }
  }
>
  <div
    className="row login-row"
  >
    <div
      className="image col-lg col-xl"
    />
    <div
      className="col-lg col-xl login-section"
    >
      <div
        className="logo"
      >
        <img
          alt="unicef vietnam"
          className="moh"
          src="test-file-stub"
        />
        <img
          alt="unicef vietnam"
          className="unicef"
          src="test-file-stub"
        />
      </div>
      <div
        className="center"
      >
        <h3>
          MIECD VIETNAM
        </h3>
        <ProviderLinks
          providers={
            Object {
              "onadata": Object {
                "accessTokenUri": "https://stage-api.ona.io/o/token/",
                "authorizationUri": "https://stage-api.ona.io/o/authorize/",
                "clientId": "hunter2",
                "redirectUri": "https://example.com/oauth/callback/onadata/",
                "scopes": Array [
                  "read",
                  "write",
                ],
                "state": "abc",
                "userUri": "https://stage-api.ona.io/api/v1/user.json",
              },
            }
          }
        >
          <div
            className="gatekeeper-login"
          >
            <p
              className="gatekeeper-p"
            >
              Please log in with one of the following providers
            </p>
            <p
              className="gatekeeper-p item"
              key="onadata"
            >
              <a
                className="gatekeeper-btn"
                href="https://stage-api.ona.io/o/authorize/?client_id=hunter2&redirect_uri=https%3A%2F%2Fexample.com%2Foauth%2Fcallback%2Fonadata%2F&scope=read%20write&response_type=token&state=abc"
              >
                onadata
              </a>
            </p>
          </div>
        </ProviderLinks>
      </div>
    </div>
  </div>
</CustomOauthLogin>
`;
